### PR TITLE
Replace replace

### PR DESCRIPTION
**Pull request by dhendo**
Re-written to behave more like the Jinja2 implementation - it calls directly into python's replace().
- Now does not result in an infinite loop if no maxCount is used and the old string is a substring of replacement. fixes mozilla/nunjucks#201
- Handle numbers a bit better
- Improved tests
- Replacing an empty string will interleave the replacement

### CODE REVIEW

File: src/filters.js [hunk 1]
```diff
@@ -191,19 +191,51 @@ var filters = {
...
+        if(typeof str === 'number'){
+            str = str +"";
+        }
+
+        if(typeof old !== "string") {
+            return str;
+        }
+
+        // ShortCircuits
+        if(!old){
```
Comment (jlongster) [inline src/filters.js]: These `if` statements are confusing. Above you say if old is not a string, then return `str`, but here you are treating it like a boolean. If you mean to check for the empty string, please do a strict check like `old !== ''`. Can you combine the above `if` statements into `if/else` where that works too?

**Comment by dhendo** [comment]
@jlongster Squished the ifs where possible. Updated empty string check and added some more tests.

**Comment by luto** [comment]
@dhendo thank you for staying on this :)

**Comment by dhendo** [comment]
Added more tests and a fix for when the input is not a string (e.g. undefined)

**Comment by jlongster** [comment]
@dhendo so sorry it's taken so long to respond. can you address the test failures on travis? why is that failing?

**Comment by dhendo** [comment]
It would have been helpful if I'd committed both files the first time round!

**Comment by jlongster** [comment]
@dhendo yeesh I merged a bunch of other stuff and now it can't merge :/ can you rebase? sorry!

**Comment by dhendo** [comment]
The rebase looks to have gone a bit awry - I'll re-apply on top of a clean branch and create a new PR.

**Comment by dhendo** [comment]
https://github.com/mozilla/nunjucks/pull/420

**Pull request closed without merging**